### PR TITLE
feat: Add API key generation to CDK

### PR DESCRIPTION
**Pull request by AshtonStephens**
## Description

Closes: #820

## Changes

Adds the creation of API keys.

## Testing Information

Tested locally by deploying the API with other changes.

Builds upon:
- #816 
- #819 

## Checklist:

- [x] I have performed a self-review of my code
- [x] My changes generate no new warnings
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: emily/cdk/lib/emily-stack-utils.ts [hunk 1]
```diff
@@ -77,6 +82,16 @@ export class EmilyStackUtils {
         return this.tablesOnly;
     }
 
+    /*
+     * Returns the number of signer API keys to create.
+     */
+    public static getNumSignerApiKeys(): number {
+        this.numSignerApiKeys ??= parseInt(process.env.NUM_SIGNER_API_KEYS ?? (Constants.DEFAULT_NUM_SIGNER_API_KEYS).toString());
+        if (this.numSignerApiKeys === undefined) {
+            throw new Error('Must define number of signer API');
```
Comment (Jiloc) [inline emily/cdk/lib/emily-stack-utils.ts:91]:
```suggestion
            throw new Error('Must define number of signer API keys');
```

**Review by Jiloc — COMMENTED** [review]
I don't have much experience with CDK, but reviewing the PR, everything seems in order to me! I just made a small suggestion for a slightly more descriptive error message

File: emily/cdk/lib/emily-stack.ts [hunk 2]
```diff
@@ -279,14 +279,46 @@ export class EmilyStack extends cdk.Stack {
...
 
+        // Create a usage plan that will be used by the Signers. This will allow us to throttle
+        // the general API more than the signers.
+        const signerApiUsagePlanId: string = `SignerApiUsagePlan`;
+        const signerApiUsagePlan = api.addUsagePlan(signerApiUsagePlanId, {
+            name: EmilyStackUtils.getResourceName(signerApiUsagePlanId, props),
+            throttle: {
+                // These are very high limits. We can adjust them down as needed.
+                rateLimit: 100,
+                burstLimit: 200,
```
Comment (djordon) [inline emily/cdk/lib/emily-stack.ts:290]: Are these requests per second?

Comment (AshtonStephens) [reply emily/cdk/lib/emily-stack.ts:290]: Yes, we don't expect signers to get anywhere near this iirc.

File: emily/cdk/lib/emily-stack-utils.ts (hunk 1 above)
Comment (AshtonStephens) [reply emily/cdk/lib/emily-stack-utils.ts]: Done!

**Review by djordon — APPROVED** [review]
Looks good, just one question.

File: emily/cdk/lib/constants.ts [hunk 3]
```diff
@@ -24,4 +24,9 @@ export class Constants {
      * Default stage name used when no stage name is provided.
      */
     static DEFAULT_STAGE_NAME: string = "dev";
+
+    /**
+     * Default number of signer API keys to create.
+     */
+    static DEFAULT_NUM_SIGNER_API_KEYS: number = 2;
```
[the remark this answers is not in the document]
Comment (AshtonStephens) [reply emily/cdk/lib/constants.ts:31]: Why not 2?

[the remark this answers is not in the document]
Comment (AshtonStephens) [reply emily/cdk/lib/constants.ts:31]: The defaults are all for a dev environment which means that for production everything needs to be specified manually

[the remark this answers is not in the document]
Comment (djordon) [reply emily/cdk/lib/constants.ts:31]: Yeah for a dev environment we'd want one for each signer, which currently is set to 3.

[the remark this answers is not in the document]
Comment (AshtonStephens) [reply emily/cdk/lib/constants.ts]: Done.

**Review by Jiloc — APPROVED** [review]
All good!

**Pull request merged**